### ISSUE TITLE
Memoize not working properly with `NaN`

### ISSUE THREAD

**Issue by ohoho7** [alexreardon/memoize-one#101]
Doc has statement:
Tested with all built in JavaScript types.
But it is not working properly with NaN value, example:
```
let i = 0;
function getIncrementedNumber(parameter) {
    return i++;
}
const memoizedIncrementedNumber = memoize(getIncrementedNumber);
const value = NaN;
console.log(memoizedIncrementedNumber(value)); // log 0
console.log(memoizedIncrementedNumber(value)); // log 1
```

**Comment by alexreardon** [comment]
We will probably need to add an `NaN` check. `NaN` !== `NaN`

**Comment by alexreardon** [comment]
this check is not correct for `NaN`:

```
for (let i = 0; i < newInputs.length; i++) {
    // using shallow equality check
    if (newInputs[i] !== lastInputs[i]) {
      return false;
    }
  }
```

**Comment by alexreardon** [comment]
Sorry for my delay in this. I have been overwhelmed with other things. I want to get this in, but I want to do some benchmarking of different approaches first

**Comment by alexreardon** [comment]
Closed by #112
